### PR TITLE
Update Feature: spare parts 

### PR DESCRIPTION
**Pull request by choubchanthou**

### CODE REVIEW

File: lib/Elements/Product.php [hunk 1]
```diff
@@ -95,4 +97,60 @@ public function getBrand()
...
+    //Override save
+    public function save()
+    {
+        $this->loadOriginal();
+
+        if ($this->isPersisted()) {
+            if (static::canUpdate()) {
+                $spare_parts = $this->getElementBody(false)->spare_parts;
+                $productId = $this->id;
+                if ($spare_parts !== []) {
```
Comment (salysakey) [inline lib/Elements/Product.php:110]: Should check array size instead of comparing 2 arrays.

Comment (salysakey) [reply lib/Elements/Product.php:110]: Please also prevent `null` vaue of `$spare_parts`

File: phpunit.xml [hunk 2]
```diff
@@ -3,8 +3,8 @@
     colors="auto"
     >
     <php>
-        <env name="SHOPRUNBACK_URL" value="https://sandbox.dashboard.shoprunback.com"/>
-        <env name="SHOPRUNBACK_TOKEN" value="1yHRJJFgzU4McVsuicLNbtm5sPpYeEW2eCpt5-awiWRx7218wg"/>
+        <env name="SHOPRUNBACK_URL" value="https://staging.dashboard.shoprunback.com"/>
+        <env name="SHOPRUNBACK_TOKEN" value="BnnuvaVbKm9WhFnwERpLVxaJZeSAZLYidvc6sWqfzzCqhzynRA"/>
```
Comment (salysakey) [inline phpunit.xml:7]: Should not put a token in the code, especially a public library.

File: tests/Elements/SparePartTrait.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public static function getElementClass()
+    {
+        return 'Shoprunback\Elements\SparePart';
+    }
+
+    public static function createDefault()
+    {
+        $name = self::randomString();
+        $reference = self::randomString();
```
Comment (salysakey) [inline tests/Elements/SparePartTrait.php:17]: These 2 variables are not necessary. 
`$sparePart->name = self::randomString();`

File: phpunit.xml (hunk 2 above)
Comment (hamlyPov) [reply phpunit.xml:7]: its a testing token and its already put in env for unittest.

File: tests/Elements/SparePartTrait.php (hunk 3 above)
Comment (hamlyPov) [reply tests/Elements/SparePartTrait.php:17]: Its for unittest to generate value for varaible.

**Review by salysakey — APPROVED** [review]

**Pull request merged**